runtime/sema/type.go

```diff
@@ -5769,31 +5768,28 @@ func (t *InterfaceType) MemberMap() *StringMemberOrderedMap {
 }
 
 func (t *InterfaceType) SupportedEntitlements() (set *EntitlementOrderedSet) {
-	supportedEntitlements := t.supportedEntitlements
-	if supportedEntitlements != nil {
-		return supportedEntitlements
-	}
+	t.supportedEntitlementsOnce.Do(func() {
+		set = orderedmap.New[EntitlementOrderedSet](t.Members.Len())
```

Comment:
```suggestion
		set := orderedmap.New[EntitlementOrderedSet](t.Members.Len())
```

and change return type to `*EntitlementOrderedSet`